### PR TITLE
Enable E2E testing for v1.15 release cycle

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind failing-test

**What this PR does / why we need it**:

This PR updates main E2E config for the v1.15 dev cycle.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mboersma** [comment]
I think you need to move the e2e template dirs to match:

```shell
git mv test/e2e/data/infrastructure-azure/v1.12.2 test/e2e/data/infrastructure-azure/v1.13.2
git mv test/e2e/data/infrastructure-azure/v1.13.0 test/e2e/data/infrastructure-azure/v1.14.0
```

**Comment by nojnhuh** [comment]
> I think you need to move the e2e template dirs to match:
> 
> ```shell
> git mv test/e2e/data/infrastructure-azure/v1.12.2 test/e2e/data/infrastructure-azure/v1.13.2
> git mv test/e2e/data/infrastructure-azure/v1.13.0 test/e2e/data/infrastructure-azure/v1.14.0
> ```

Not something we have to do right now, but I think it's worth looking to see if we can remove exact versions from these paths and everywhere else that we can. Theoretically it seems like the only place they're definitely needed is the download URL, and everything else could have a generic name indicating "latest supported version" or "oldest supported version" that we don't need to bump every time we bump the actual version.

File: test/e2e/data/shared/v1beta1_provider/metadata.yaml [hunk 1]
```diff
@@ -19,3 +19,6 @@ releaseSeries:
   - major: 1
```
Comment (nojnhuh) [inline test/e2e/data/shared/v1beta1_provider/metadata.yaml:19]: If we want to stay extra tidy, we should be able to remove the 1.11 and 1.12 stanzas here.

**Review by nojnhuh — COMMENTED** [review]
/lgtm

File: test/e2e/data/shared/v1beta1_provider/metadata.yaml (hunk 1 above)
Comment (jackfrancis) [reply test/e2e/data/shared/v1beta1_provider/metadata.yaml:19]: @nawazkh could you do that as a follow-up PR once this lands?

Lemme know if you don't understand why we would do this, but tl;dr we're maintaining a static set of CAPZ version that we test in azure-dev.yaml, and we only need entries in this file for the versions that we have test configurations to actually use.

**Review by mboersma — APPROVED** [review]
/approve

**Pull request merged**